### PR TITLE
fix(calendar): doesn't handle timezone from ng-model-options

### PR DESCRIPTION
**Pull request by gaby64**
## PR Checklist
Please check that your PR fulfills the following requirements:
- [ ] The commit message follows [our guidelines](https://github.com/angular/material/blob/master/.github/CONTRIBUTING.md#-commit-message-format)
- [ ] Tests for the changes have been added or this is not a bug fix / enhancement
- [ ] Docs have been added, updated, or were not required

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Enhancement
[ ] Documentation content changes
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: 
#10431

## What is the new behavior?
- creates a date object with correct timezone offset
- checks if value is null, to not create an invalid date

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by gaby64** [comment]
@googlebot I signed it!

File: src/components/datepicker/js/datepickerDirective.js [hunk 1]
```diff
@@ -959,7 +959,7 @@
   DatePickerCtrl.prototype.onExternalChange = function(value) {
     var timezone = this.$mdUtil.getModelOption(this.ngModelCtrl, 'timezone');
 
-    this.date = value;
+    this.date = !value ? value : new Date(this.locale.formatDate(value, timezone));
```
Comment (Splaktar) [inline src/components/datepicker/js/datepickerDirective.js]: For clarity, perhaps this is better?
```js
this.date = angular.isDate(value) ? new Date(this.locale.formatDate(value, timezone)) : value;
```

**Comment by Splaktar** [comment]
@gaby64 Thank you for making your first contribution.

Can you please address the the review feedback?
- fix commit message
- respond to suggested changes to code
- add a unit test

If you won't be able to do this or have any questions, please let me know.

**Comment by gaby64** [comment]
sorry, I wont be able to, I have migrated to vuejs

**Comment by Splaktar** [comment]
I tested these changes locally and they don't appear to be sufficient to resolve this issue.

**Pull request closed without merging**